core/arch/aarch64/emit_utils.c

```diff
@@ -605,8 +604,7 @@ append_restore_simd_reg(dcontext_t *dcontext, instrlist_t *ilist, bool absolute)
                 INSTR_CREATE_ldr(
                     dcontext, opnd_create_reg(DR_REG_P0 + i),
                     opnd_create_base_disp(
-                        DR_REG_X1, DR_REG_NULL, 0,
-                        i * (proc_get_vector_length_bytes() / 8),
+                        DR_REG_X1, DR_REG_NULL, 0, i * sizeof(dr_simd_t),
```

Comment:
Lines `634` and `848` also load `P15` as part of saving/restoring the `FFR` register so those need to be updated too.

`/15 * (proc_get_vector_length_bytes() / 8)/15 * sizeof(dr_simd_t)/`

I think this is what is causing the `P15` corruption errors in the scatter/gather test.

Reply:
Done.

Reply:
PTAL